parser.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+// Parser Structure: dictionary, sections
+// It's basically a map of a map, map of sections where each section maps to keys : values
+// map[section] --> returns a map where each key --> value
+type Parser struct {
+	dictionary map[string]map[string]string
+}
+
+// NewParser creates an instance of our Parser, which is iniitially an empty map of maps
+func NewParser() Parser {
+	return Parser{make(map[string]map[string]string)}
+}
+
+// ErrCouldNotOpen happens when file cannot be opened and provides file name
+var ErrCouldNotOpen error
+
+// ErrMissingValueAssignment happens when a key isn't followed by an = statement
+var ErrMissingValueAssignment error
+
+// ErrSectionNameMissingClosure happens when section name is missing the ] paranthesis
+var ErrSectionNameMissingClosure error
+
+// ErrWrongParanthesisOrder happens when section name starts by the wrong paranthesis ']'
+var ErrWrongParanthesisOrder error
+
+// ErrInvalidSectionName happens when section is written in a wrong form --> ex: sectionName]
+var ErrInvalidSectionName error
+
+// ErrCouldNotWriteToFile happens when file cannot be written to
+var ErrCouldNotWriteToFile error
+
+// LoadFromFile loads ini file
+// Saves all lines locally into an array of strings
+func (parsedMap *Parser) LoadFromFile(fileName string) error {
+	var iniLines []string
+	var input io.Reader
+
+	file, err := os.Open(fileName)
+	if err != nil {
+		ErrCouldNotOpen = fmt.Errorf("cannot open file: %s", fileName)
+		return ErrCouldNotOpen
+	}
+	defer file.Close()
+
+	input = file
+	scanner := bufio.NewScanner(input)
+	for scanner.Scan() {
+		iniLines = append(iniLines, scanner.Text())
+	}
+	return parsedMap.parserLogic(iniLines)
```

Comment:
> parserLogic is not a good name

name it a helper function? or parsingHelper ??

Reply:
parseIni, parseLines or somthing, or just move the logic to LoadFromString Function

Reply:
> parseIni, parseLines or somthing, or just move the logic to LoadFromString Function

okay I'll rename it, because both loadfromstring and loadfromfile use it, so it'l be repetitive code which isn't good
fa that's why I made the logic in a separate function

Reply:
you know can just call LoadFromString from LoadFromFile without any repetitive code no?

Reply:
> mString from LoadFromFile without any repetitive code no?

strings still split the string so it has an extra step from loadfromfile